### ISSUE TITLE
Android arm平台 Llama-2-7b-chat-MNN输出乱码

### ISSUE THREAD

**Issue by xiaoxiaoyuwen** [wangzhaode/mnn-llm#204]
- 模型
  - https://modelscope.cn/models/zhaode/Llama-2-7b-chat-MNN
  - PS：该模型special token id 有个32000会导致越界崩溃
- 代码
  - master 分支 commit id = 1f2a68d5bfd020b9000b45f061d508a22cf1d2a7
- 问题
  - 使用 script/android_build.sh 脚本编译出的cli_demo在Android  arm环境运行输出乱码
  - 使用 script/build.sh脚本编译出的cli_demo在Linux X86 环境正常
  - 测试命令 ./cli_demo Llama-2-7b-chat-MNN/config.json

- 问题定位
  -  Android环境下 (logits) VARP readMap 和writeMap 获取到的数据，打印出来不一致
  -  sample函数使用writeMap 对分数做处理后获取的argmax不正确导致乱码
  ```
        // llm.cpp forward函数 验证readMap和writeMap
        auto dump_fp = [](VARP var, const char* name) {
            auto ptr = var->readMap<float>();
            auto size = var->getInfo()->size;
            printf("%s = [", name);
            for (int i = 0; i < 5; i++) printf("%f, ", ptr[i]);
            printf(" ..., ");
            for (int i = size - 5; i < size; i++) printf("%f, ", ptr[i]);
            printf("]\n");
        };
        auto dump_fp2 = [](VARP var, const char *name) {
          auto ptr = var->writeMap<float>();  // 从writeMap获取数据
          auto size = var->getInfo()->size;
          printf("%s = [", name);
          for (int i = 0; i < 5; i++)
            printf("%f, ", ptr[i]);
          printf(" ..., ");
          for (int i = size - 5; i < size; i++)
            printf("%f, ", ptr[i]);
          printf("]\n");
        };
        ....
        auto outputs = modules_.back()->onForward({hidden_states, attention_mask, position_ids, past_key_values_[0]});
        ExecutorScope::Current()->gc(Executor::FULL);
        logits = outputs[0];
        dump_fp(logits, "logits from readMap");
        dump_fp2(logits, "logits from writeMap");
        past_key_values_[0] = outputs[1];
  ```
   
   输出
   ```
   // android环境输出
  logits from readMap = [-3.685547, 2.187500, 6.539062, -0.942383, -1.372070,  ..., -3.001953, -0.791504, 1.873047, -2.328125, -2.080078, ]
  logits from writeMap = [3.511925, -0.004220, 0.000059, 0.000000, 17.896412,  ..., 0.000000, 0.000000, 0.000000, 0.000000, 0.000000, ]
  
  // linux 环境输出
  logits from readMap = [-3.734469, 1.471044, 5.270651, -1.283450, -1.496580,  ..., -3.019870, -0.337005, 1.685822, -2.157527, -2.499675, ]
   logits from writeMap = [-3.734469, 1.471044, 5.270651, -1.283450, -1.496580,  ..., -3.019870, -0.337005, 1.685822, -2.157527, -2.499675, ]
   ```

**Comment by wangzhaode** [comment]
收到🫡，后续测试一下
